es-ds-components/components/es-button.vue

```diff
@@ -0,0 +1,95 @@
+<script setup lang="ts">
+// Cannot import as Button due to issue with native element Button
+import PrimeButton from 'primevue/button';
+
+/*
+severity:
+
+Prime severity (variant="dark-bg" in particular) doesn't map to
+"success" | "help" | "secondary" | "info" | "warning" | "danger" | "contrast"
+
+size:
+Prime size "small" | "large" doesn't map to our 3-sizes in the design-system
+
+icon usage:
+Prime provides icon & loadingIcon slots which can hook into the loading property,
+currently we're not using this (but we may want to re-visit)
+*/
+
+const props = defineProps({
+    /**
+        * Button Variant
+        *
+        */
+    variant: {
+        type: String,
+        default: 'primary',
+        required: false,
+    },
+    /**
+        * Outline only button
+        */
+    outline: {
+        type: Boolean,
+        default: false,
+    },
+    /**
+        * Size of button
+        */
+    size: {
+        type: String,
+        validator: (val) => ['lg', 'md', 'sm'].includes(String(val)),
+        default: 'md',
+    },
+    /**
+        * Only works for 'link' variant buttons.
+        * Removes the fixed height and padding so the button can fit nicely within a block of text.
+        */
+    inline: {
+        type: Boolean,
+        default: false,
+    },
+
+    // PrimeVue Arguments with es-ds defaults
+    rounded: {
+        type: Boolean,
+        default: true,
+    }
+})
+
+const getRootClasses = computed(() => {
+    const withOutline = props.outline ? '-outline' : '';
+    const btnVariant = props.variant === 'link' ? 'btn-link' :
+        props.variant === 'dark-bg' ? `btn${withOutline}-dark-bg` :
+            `btn${withOutline}-primary`;
+    const btnInline = props.inline ? 'inline' : '';
+    const btnSize = props.size ? `btn-${props.size }` : 'btn-md';
+    return `btn ${btnVariant} ${btnInline} ${btnSize}`;
+});
+
+const isLink = computed(() => props.variant === 'link');
+</script>
+
+<template>
+    <prime-button
+        :pt="{
+            root: { class: getRootClasses }
+        }"
+        v-bind="$attrs"
+        :link="isLink"
+        :outlined="outline">
```

Comment:
Nit: `:link` and `:outlined` aren't doing anything since our classes already handle that. v-bind is also unecessary since [fallthrough attributes are implicit](https://vuejs.org/guide/components/attrs) when there's only one child component

Reply:
Good catch, done via [802ad9e](https://github.com/EnergySage/es-ds/pull/1487/commits/802ad9e1bb8a806428fe19bc83fb80b71ccc657a), [cb351ea](https://github.com/EnergySage/es-ds/pull/1487/commits/cb351eaa333f86ca7620e113b21bae664169b5f3), and [8701f7c](https://github.com/EnergySage/es-ds/pull/1487/commits/8701f7c87a40bb77a7921d24cb041d6a9a76e6a3)